src/main/java/teammates/common/util/Url.java

```diff
@@ -2,14 +2,12 @@
 
 public class Url {
 
+    protected final String baseUrl;
     protected String urlString;
 
-    public Url(String url) {
-        this.urlString = url.replace(getAppUrl(), ""); // force the URL to be relative
-    }
-
-    protected String getAppUrl() {
-        return Config.APP_URL;
+    public Url(String baseUrl, String url) {
```

Comment:
When adding defensive code, we can defend against subtle errors other developers are likely to commit. e.g. trailing slash, `baseUrl` does not match the base of the `url`. These relate to implicit rules other TM developers are "supposed to" follow, but it is possible someone might miss them. On the other hand, we might not have to defend (no harm doing so though) against things like random strings being passed instead of URLs.